src/atomic64.rs

```diff
@@ -205,9 +208,24 @@ impl Atomic for AtomicU64 {
 }
 
 impl AtomicU64 {
-    /// Get the value with the provided memory ordering.
-    pub fn compare_and_swap(&self, current: u64, new: u64, ordering: Ordering) -> u64 {
-        self.inner.compare_and_swap(current, new, ordering)
+    /// Stores a value into the atomic integer if the current value is the same
+    /// as the current value.
+    ///
+    /// This function is allowed to spuriously fail even when the comparison
+    /// succeeds, which can result in more efficient code on some platforms. The
+    /// return value is a result indicating whether the new value was written
+    /// and containing the previous value.
+    ///
+    /// See [`StdAtomicU64`] for details.
+    pub fn compare_exchange_weak(
```

Comment:
On this one, I think you'd want to expose the non-weak version.

Reply:
`AtomicU64::compare_exchange_weak` is only used in `HistogramCore::proto`. As `HistogramCore::proto` is trying to compare and set a value in a loop it should suffice to use `compare_exchange_weak` instead of `compare_exchange`.

If I am not mistaken `AtomicU64` is not accessible when used in a metric like `GenericCounter`, thus I don't think we need to expose `compare_exchange` as well in order to maintain backward-compatibility.

Does the above make sense @lucab?

Reply:
My line of reasoning was that it is a public type+method, so if people are using this for whatever reason, then they'd want to have a 1-to-1 replacement when upgrading to the new version.

As it was just recently introduced by your atomic-histogram PR, likely nobody is yet using this, thus I agree it is fine to mark this a `pub(crate)` and keep the weak version.